### PR TITLE
Add credo and fix up credo recommendations

### PR DESCRIPTION
**Pull request by paulruescher**
# Status
**READY**

# Description
Adding credo to the project. 

Some of the changes are a bit weird. For example, credo expects that a raw value starts a pipe, so you'll see scenarios like https://github.com/mweibel/facebook.ex/compare/master...paulruescher:add-credo?expand=1#diff-9944d4d916ca0f3be718c57f915ae4d6R298

Otherwise, things seem a bit more clear already.

Take a look at https://github.com/rrrene/credo#configuration if you're interested in seeing the configs we could setup. I'm generally OK with the defaults for now.

# Considerations
Can we get `MIX_ENV=test mix quality` run in travis now? Usually I'd just add to the `.travis.yml` file, but I'm not sure how you have things setup at the moment.

### CODE REVIEW

**Comment by paulruescher** [comment]
yeah, I agree that they looked a little off heh. I'll look into changing up the `@types` or maybe some other alternative

**Comment by paulruescher** [comment]
Updated per convo. @mweibel got any feedback on how to get this running on travis? Did you setup the instructions within Travis?

**Comment by mweibel** [comment]
The raw values starting with a pipe do look awkward to me, but then again I don't really mind. 
specs/types: some functions' params have the type directly in the spec, some have it's own type (even if it's a simple type like `atom`) now.. Should we unify that or what's your opinion? I guess if you look at the hexdocs in the end it doesn't really matter.

@paulruescher travis: At the moment we just use the default command which travis sets up automatically for elixir projects if nothing is specified (https://docs.travis-ci.com/user/languages/elixir/).
I guess you can just add the following to travis and it should work:
```
script:
  - mix test
  - mix quality # do we need MIX_ENV here? probably, but unsure if travis sets that automatically
```

**Comment by paulruescher** [comment]
> The raw values starting with a pipe do look awkward to me, but then again I don't really mind.

I agree, but I think it's mostly because it's the sigils that throw me off. If it was a variable I wouldn't mind.

> specs/types: some functions' params have the type directly in the spec, some have it's own type (even if it's a simple type like atom) now.. Should we unify that

Hmm, I didn't think about that. I think I'm OK inlining types where possible, but it might be nice do the ones that are multiples for consistencies sake.

> I guess you can just add the following to travis and it should work:

No better way to find out than to try it, hey?

File: lib/facebook.ex [hunk 1]
```diff
@@ -22,17 +23,24 @@ defmodule Facebook do
...
     ]
 
     supervise(children, strategy: :one_for_one)
   end
 
-  @type fields :: list
-  @type access_token :: String.t
-  @type response :: {:ok, Map.t} | {:error, Map.t}
   @type using_appsecret :: boolean
+  @type react_type :: atom
```
[the remark this answers is not in the document]
Comment (mweibel) [reply lib/facebook.ex]: That's ok I guess. Could also order alphabetically but in the end it doesn't really matter TBH.

**Comment by mweibel** [comment]
That all looks good to me. Should I merge or is there anything open? 
Also if it's ok I'll release a new version afterwards. In case other things come in we can release those afterwards, what do you think?

**Comment by paulruescher** [comment]
I think it's good to merge.

As for a version bump, I'd ask @tfinnell. We have a bunch of breaking changes because of my last PR. I think most of the stuff we're thinking of adding is either behind the public API, or additional features. But I'm not positive.

**Comment by tfinnell** [comment]
A new version is fine afaic.

**Comment by mweibel** [comment]
Great, thanks you two! I just released 0.16.0 on hex.

**Pull request merged**